src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -42,58 +43,84 @@ public boolean contains(Person toCheck) {
      */
     public String getNonUniqueAttributeType(Person toCheck) {
         requireNonNull(toCheck);
-        StringBuilder stringBuilder = new StringBuilder();
+        ArrayList<String> nonUniqueAttributeTypes = new ArrayList<>();
         for (int i = 0; i < internalList.size(); i++) {
             Person otherPerson = internalList.get(i);
             if (otherPerson.isSamePerson(toCheck)) { // Unique attributes already exist in the address book.
                 if (otherPerson.getPhone().equals(toCheck.getPhone())) {
-                    appendPhoneAttributeToString(stringBuilder);
-                }
-                if (otherPerson.getMatriculationNumber().equals(toCheck.getMatriculationNumber())) {
-                    appendMatriculationNumberAttributeToString(stringBuilder);
+                    addNonUniqueAttributeType(nonUniqueAttributeTypes, 0);
                 }
                 if (otherPerson.getEmail().equals(toCheck.getEmail())) {
-                    appendEmailAttributeToString(stringBuilder);
+                    addNonUniqueAttributeType(nonUniqueAttributeTypes, 1);
+                }
+                if (otherPerson.getMatriculationNumber().equals(toCheck.getMatriculationNumber())) {
+                    addNonUniqueAttributeType(nonUniqueAttributeTypes, 2);
                 }
             }
         }
-        return stringBuilder.toString();
+        return getNonUniqueAttributeTypesMessage(nonUniqueAttributeTypes);
     }
 
     /**
-     * Appends the necessary syntax for the Phone attribute into the non-unique attribute type string.
-     * @param stringBuilder Non-unique attribute type string
+     * Adds a non-unique attribute type to the list if it is not yet added.
+     *
+     * @param nonUniqueAttributeTypes   The list containing the non-unique attribute types
+     * @param uniqueAttributeTypesIndex The index of the unique attributes specified by the {@code Person} class
      */
-    private void appendPhoneAttributeToString(StringBuilder stringBuilder) {
-        if (!stringBuilder.toString().contains("Phone")) {
-            stringBuilder.append("Phone");
+    private void addNonUniqueAttributeType(ArrayList<String> nonUniqueAttributeTypes, int uniqueAttributeTypesIndex) {
+        assert uniqueAttributeTypesIndex >= 0 && uniqueAttributeTypesIndex <= 2 : false;
+        String attributeType = Person.UNIQUE_ATTRIBUTE_TYPES[uniqueAttributeTypesIndex];
+        if (!nonUniqueAttributeTypes.contains(attributeType)) {
+            nonUniqueAttributeTypes.add(attributeType);
         }
     }
 
     /**
-     * Appends the necessary syntax for the Matriculation Number attribute into the non-unique attribute type string.
-     * @param stringBuilder Non-unique attribute type string
+     * Returns the message for non-unique attribute types. This message needs further processing as there is a
```

Comment:
Just my opinion but maybe instead of saying "This message needs further processing...", it can be "{@code Person#removeWhiteSpaceBeforeComma(StringBuilder)} is applied on the message to remove any extra spacings when using punctuations".  Up to you to change!

Reply:
Thanks for pointing this out, I have changed the implementation of this method and have removed the method for removing the white space.